cmd/transformEventsOnDemand.go

```diff
@@ -0,0 +1,134 @@
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var blockToTransform int
+
+// transformEventsOnDemandCmd represents the transformEventsOnDemand command
+var transformEventsOnDemandCmd = &cobra.Command{
+	Use:   "transformEventsOnDemand",
+	Short: "Executes transformers for a given block",
+	Long:  ``,
+	Run: func(cmd *cobra.Command, args []string) {
+		SubCommand = cmd.CalledAs()
+		LogWithCommand = *logrus.WithField("SubCommand", SubCommand)
+
+		executeOnDemand(int64(blockToTransform))
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(transformEventsOnDemandCmd)
+	transformEventsOnDemandCmd.Flags().IntVarP(&blockToTransform, "block-to-transform", "b", 0, "block to transform events for on demand")
+}
+
+func executeOnDemand(blockNumber int64) {
+	configErr := prepConfig()
+	if configErr != nil {
+		LogWithCommand.Fatalf("failed to prepare config: %s", configErr.Error())
+	}
+	// Get the plugin path and load the plugin
+	_, pluginPath, pathErr := genConfig.GetPluginPaths()
+	if pathErr != nil {
+		LogWithCommand.Fatalf("failed to get plugin paths: %s", pathErr.Error())
+	}
+
+	LogWithCommand.Info("linking plugin ", pluginPath)
+	plug, openErr := plugin.Open(pluginPath)
+	if openErr != nil {
+		LogWithCommand.Fatalf("linking plugin failed: %s", openErr.Error())
+	}
+
+	// Load the `Exporter` symbol from the plugin
+	LogWithCommand.Info("loading transformers from plugin")
+	symExporter, lookupErr := plug.Lookup("Exporter")
+	if lookupErr != nil {
+		LogWithCommand.Fatalf("loading Exporter symbol failed: %s", lookupErr.Error())
+	}
+
+	// Assert that the symbol is of type Exporter
+	exporter, ok := symExporter.(Exporter)
+	if !ok {
+		LogWithCommand.Fatal("plugged-in symbol not of type Exporter")
+	}
+
+	// Use the Exporters export method to load the EventTransformerInitializer, StorageTransformerInitializer, and ContractTransformerInitializer sets
+	ethEventInitializers, _, _ := exporter.Export()
+
+	// Setup bc and db objects
+	blockChain := getBlockChain()
+	db := utils.LoadPostgres(databaseConfig, blockChain.Node())
+
+	// Execute over transformer sets returned by the exporter
+	// Use WaitGroup to wait on the goroutines
+	if len(ethEventInitializers) > 0 {
+		// Setting maxUnexpectedErrors to 1 so that we don't continue looping
+		maxUnexpectedErrors = 1
+
+		currentCheckCount, err := getCurrentCheckCount(blockNumber, &db)
+		if err != nil {
+			panic("Failed to get current check_count from the headers table")
+		}
+
+		extractor := logs.NewLogExtractor(&db, blockChain)
+		delegator := logs.NewLogDelegator(&db)
+		ew := watcher.NewEventWatcher(&db, blockChain, extractor, delegator, maxUnexpectedErrors, retryInterval)
+		addErr := ew.AddTransformers(ethEventInitializers)
+
+		// Overriding the normal behavior to just transform one block
+		extractor.OverrideStartingAndEndingBlocks(&blockNumber, &blockNumber)
+		extractor.OverrideRecheckHeaderCap(currentCheckCount)
+		ew.UnsetExpectedExtractorError()
```

Comment:
These 3 lines are the crux of the changes that make this command work for just one block. 
- `extractor.OverrideStartingAndEndingBlocks` allows us to set the starting and ending blocks for the extractor, regardless of the ones that are configured for the transformers so that we can narrow down the scope of what this command does - just transforms one block's worth of logs. 
- `extractor.OverrideRecheckHeaderCap` allows us to check a header more than the constants recheckHeaderCap of 5. It can be set to anything, but currently I'm fetching the check_count on the current header, and then checking it one more time.
- `ew.UnsetExpectedExtractorError()` removes the expected NoUncheckedHeaders error so that once the header is checked that one more time (or however we choose to configure the recheckHeaderCap) the watcher will return an error and the command exits. This was done to make sure that the command doesn't loop forever without doing anything productive.